nanopub/publication.py

```diff
@@ -121,83 +100,119 @@ def from_assertion(cls, assertion_rdf: rdflib.Graph,
                                  'npx:introduces predicate, so you can not also use the '
                                  'introduces_concept argument')
 
-        # To be replaced with the published uri upon publishing
-        this_np = rdflib.Namespace(DUMMY_NANOPUB_URI + '#')
+    @classmethod
+    def from_assertion(cls, assertion_rdf: rdflib.Graph,
```

Comment:
I completely refactored `from_assertion` method in an attempt to reduce the complexity. I actually really like the result, so I suggest to also let the build fail on complexity violations.

Reply:
I agree this method is much cleaner now, good job! I have no _strong_ objection to making complexity a fail condition, but in my experience it can sometimes be a bit cumbersome and break the flow. However, clearly we've been ignoring it until now, so let's indeed make it a fail unless/until it becomes too annoying. I think we should keep max line length a non-fail warning though, since that genuinely is not important enough to do every single PR.